packages/components/src/navigation/item/base-content.tsx

```diff
@@ -3,14 +3,19 @@
  */
 import { ItemBadgeUI, ItemTitleUI } from '../styles/navigation-styles';
 
-export default function NavigationItemBaseContent( props ) {
+import type { NavigationItemBaseContentProps } from '../types';
+
+export default function NavigationItemBaseContent(
+	props: NavigationItemBaseContentProps
+) {
 	const { badge, title } = props;
 
 	return (
 		<>
 			{ title && (
 				<ItemTitleUI
 					className="components-navigation__item-title"
+					// @ts-expect-error The only allowed variant is `muted` ?
```

Comment:
Not really sure about what's the best change here.

Reply:
I took a deeper look. It looks like the `variant` prop is used by the Text component to add a style object that should be named as the variant:

https://github.com/WordPress/gutenberg/blob/e3c8a6cb0f7890f21d688be28fe4ead1b11a1bc0/packages/components/src/text/hook.js#L122

Looking at the [`styles`](https://github.com/WordPress/gutenberg/blob/19efda38e0f4c6784863162f3482b685a9519059/packages/components/src/text/styles.js), I can't see any exported object that would match the `body.small` key.

So, my conclusion is that `variant="body.small"` isn't actually making any difference on the way the item is rendered. To confirm my suspicion, I also tried to remove the `variant` prop from `ItemTitleUI`, and noticed no difference in the rendered output.

I therefore went ahead and removed the prop in https://github.com/WordPress/gutenberg/pull/48742/commits/d721e6bcad0732c6ccc925837e5e3032fb9aeeb4.

Let me know if this makes sense to you, or if I should make further amends